src/State/Pagination/Pagination.php

```diff
@@ -219,13 +219,38 @@ private function getGraphQlEnabled(?Operation $operation): bool
         return $operation?->getPaginationEnabled() ?? $enabled;
     }
 
+    /**
+     * Extract pagination parameter
+     * page[page] => $contextFilters['page'] with default configuration page_parameter_name: page
+     * page[number] => $contextFilters['_page'][number] with configuration page_parameter_name: page[number].
+     */
+    private function extractParameter(array $contextFilters, string $parameterName): mixed
+    {
+        preg_match_all("/[\w-]+/", $parameterName, $matches);
```

Comment:
I don't understand the need of this regex here, we know the possible parameter names no? I'd love a test to understand this better.